cmd/antrea-agent/agent.go

```diff
@@ -250,9 +252,20 @@ func run(o *Options) error {
 		isChaining,
 		routeClient,
 		networkReadyCh)
-	err = cniServer.Initialize(ovsBridgeClient, ofClient, ifaceStore, entityUpdates)
-	if err != nil {
-		return fmt.Errorf("error initializing CNI server: %v", err)
+
+	cniPodInfoStore := cnipodcache.NewCniPodInfoStore()
+
+	if features.DefaultFeatureGate.Enabled(features.SecondaryInterfaceSupport) {
+		err = cniServer.InitializeWithCniCache(ovsBridgeClient, ofClient, ifaceStore, entityUpdates, cniPodInfoStore)
+		if err != nil {
+			return fmt.Errorf("error initializing CNI server with cniPodInfoStore cache: %v", err)
+		}
+
+	} else {
+		err = cniServer.Initialize(ovsBridgeClient, ofClient, ifaceStore, entityUpdates)
```

Comment:
I think we can reuse the same Initialize() func, but based on whether the cniPodInfoStore argument is nil, the func can know it needs to initialize CNI cache or not.

Reply:
introduce another argument (and passing it nil for non secondary interface flow) impact significant of cniserver specific existing test code (also calls these Initialize function). To avoid additional changes, I spilt it separate function with cache.

Reply:
Do not get why that is a problem. You can just pass false/nil for existing callers inc. test code. Why it is significant impact?